core/cmd/hoverfly/main.go

```diff
@@ -283,6 +282,8 @@ func main() {
 		requestCache = nil
 	}
 
+	cfg.DisableProxyAuthoriation = *disableProxyAuth
```

Comment:
I think there should be a one-to-one mapping between the `hoverctl` and `hoverfly` flags because it simplifies things from the users point of view. I actually remember having issues around this last week (the certificate flag being named differently across the ctl and binary). If you only want to implement it in Hoverfly for now then go for it.

Yes I think `-proxy-auth-type` with two string values is better than the boolean.